### PR TITLE
Enhancement/swipe actions on kirby list

### LINKED ISSUE #357: [Enhancement] Add swipe actions to Kirby list (mobile web)
- It should be possible to define actions on a list item that is revealed when swiping the item. Both left and right swipe/placement should be supported.
- It should be possible to enable a "long swipe" (swiping the item over a certain threshold). Both left and right long swipe should be supported.
- If there is more than 1 action it's always the 1st action (the action closest to the side of the list) that gets triggered on long swipe (e.g. "mark" in the example screenshot below).
- It should be possible to set the background color of the revealed actions (from the Kirby progress colors: success, warning, danger).
- It should be possible to set the text and (optional) icon for each action.
- It should be possible to "switch" between different actions depending on the state of the data for each item (e.g. "Archive" vs. "Remove archive")

See:
Use case: https://app.zeplin.io/project/5c99d69a79d02b35ea3b8f41/screen/5d0770d6ea9d1b5d986e60b1
Web: https://ionicframework.com/docs/api/item-sliding

### LINKED ISSUE #382: [Enhancement] Add swipe actions to Kirby list (mobile web)
Web implementation for #357 

See:
Use case: https://app.zeplin.io/project/5c99d69a79d02b35ea3b8f41/screen/5d0770d6ea9d1b5d986e60b1
Web: https://ionicframework.com/docs/api/item-sliding
{N}: https://docs.nativescript.org/angular/ui/professional-ui-components/ng-ListView/swipe-actions

### LINKED ISSUE #383: [Enhancement] Add swipe actions to Kirby list ({N})
{N} implementation for #357 

See:
Use case: https://app.zeplin.io/project/5c99d69a79d02b35ea3b8f41/screen/5d0770d6ea9d1b5d986e60b1
Web: https://ionicframework.com/docs/api/item-sliding
{N}: https://docs.nativescript.org/angular/ui/professional-ui-components/ng-ListView/swipe-actions

### PR DESCRIPTION
**Pull request by mmmtrifork**
fixes #382 , fixes #383 , fixes #357 

- [x] #369 Must be merged first and modifications need to be added to this branch before the PR can be merged.  
- [ ] Must revise styling
- [x] Remove code that is no longer in use
- [x] Fix unit tests

### CODE REVIEW

File: src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    swipeAction.onSelected(this.item);
+    this.swipeActionSelections = {};
+  }
+
+  getIsSwipeActionSelected(swipeAction: ListSwipeAction, _: any): boolean {
+    return this.item ? this.item[swipeAction.swipeActionFlag] : false;
+  }
+
+  // Web-only
+  getIsSwipingDisabled(): boolean {
```
Comment (hafstad) [inline src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts]: Whats the point of this function?

File: src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    return false;
+  }
+
+  // Web-only
+  setList(list: any): void {
```
Comment (hafstad) [inline src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts]: Whats the point of this function?

File: src/kirby/components/list/list.component.tns.scss [hunk 3]
```diff
@@ -51,3 +51,15 @@ RadListView {
     }
   }
 }
+
```
Comment (hafstad) [inline src/kirby/components/list/list.component.tns.scss]: The styling on native doesn't seem to be that beauty, this should be looked into before merging

**Review by hafstad — CHANGES_REQUESTED** [review]
We need to put more energy into the native styling and behaviour before this can be merged

File: src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    // TODO: implement on native if needed
+  }
+
+  // Web-only
+  closeActionItems(): void {
```
[the remark this answers is not in the document]
Comment (mmmtrifork) [reply src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts]: should auto swipe the row back to its original position once we have selected a swipe option for example

File: src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts (hunk 2 above)
Comment (mmmtrifork) [reply src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts]: in order to close the swipe actions in ionic, we need to pass-in a reference to the list, hence we set it initially in the beginning, once we load the list. This doesn't apply in {N} (or at least it's not yet implemented), but the compiler will complain if we don't have the function defined in the .tns file

File: src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts (hunk 1 above)
Comment (mmmtrifork) [reply src/kirby/components/list/helpers/list-swipe-actions-helper.tns.ts]: we disable the swiper when resizing the browser, unfortunately the compiler complains if we don't have this function's definition in tns

File: src/kirby/components/list/list.component.tns.scss (hunk 3 above)
Comment (mmmtrifork) [reply src/kirby/components/list/list.component.tns.scss]: completely agree

File: src/kirby/components/list/helpers/list-swipe-action.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { ThemeColor } from '../../../helpers/theme-color.type';
+
+export interface ListSwipeAction {
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/helpers/list-swipe-action.ts]: Remember to export this in the file `../src/kirby/list.ts`, so users of kirby can import it.

File: src/kirby/components/list/list.component.scss [hunk 6]
```diff
@@ -1,7 +1,7 @@
-@import '../../scss/web-imports';
+@import "../../scss/web-imports";
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list.component.scss]: I think a replacement of `'` with `"` was made here. Better revert it.

**Review by ghost — COMMENTED** [review]
Some quick questions:

1. Has this been tested with items also having select enabled?
2. Has it been tested with sections enabled?
3. What if i have a scenario where i want some items in the list to have slide actions and some not. Is it possible we could support this? Or maybe it would better fitted for an other enhancement issue.

**Pull request closed without merging**